### PR TITLE
docs: explanation on personal search

### PR DESCRIPTION
**Pull request by alexgarel**
Adding explanations on personal search + some related APIs

### CODE REVIEW

File: docs/api/ref/api.yml [hunk 1]
```diff
@@ -430,6 +466,16 @@ components:
...
       description: '2 letter code of the language of the user. Used for localizing some fields in returned values (e.g. knowledge panels). If not passed, the language may be inferred by the Accept-Language header of the request.'
+    # same as lc above but within a path (using path template)
+    lc_in_path:
+      schema:
+        type: string
+        example: 'fr'
+      in: path
+      name: lc
+      required: true
+      description: '2 letter code of the language of the user. Used for localizing some fields in returned values (e.g. knowledge panels). If not passed, the language may be inferred by the Accept-Language header of the request.'
```
Comment (stephanegigandet) [inline docs/api/ref/api.yml:477]: If lc is not passed, it is inferred by the subdomain. There seems to be a caching bug though as I  get different results for the same request to https://fr.openfoodfacts.org/api/v2/attribute_groups

File: docs/api/ref/responses/get_attribute_groups.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          setting_name:
+            type: string
+            description: a description of the attribute to display to users
+          setting_note:
+            type: string
+            description: a complementary note on the attribute
+          default:
+            type: string
+            enum:
+            - "mandator"
```
Comment (stephanegigandet) [inline docs/api/ref/responses/get_attribute_groups.yaml:38]:
```suggestion
            - "mandatory"
```

File: docs/api/ref/schemas/product_attribute_groups.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          minimum: 0
+          maximum: 100
+          description: |
+            a numeric value for the match,
+            telling how much the products ranks well for this particular attribute.
+            The higher the value, the better the match.
+        grade:
+          description: every attribute as a grade for a to e
+          type: string
+          enum: ['a', 'b', 'c', 'd', 'e']
```
Comment (stephanegigandet) [inline docs/api/ref/schemas/product_attribute_groups.yaml:38]:
```suggestion
          enum: ['unknown','a', 'b', 'c', 'd', 'e']
```

File: html/js/product-search.js [hunk 4]
```diff
@@ -17,12 +17,14 @@
 // 		- very_good_match	score >= 75
 //		- good_match		score >= 50
 //		- poor_match		score < 50
-//		- unknown_match		at least one mandatory attribute is unknown, or unknown attributes weight more than 50% of the score
+//		- unknown_match		at least one mandatory attribute is unknown, or unknown attributes weights more than 50% of the total weights
```
Comment (stephanegigandet) [inline html/js/product-search.js:20]:
```suggestion
//		- unknown_match		at least one mandatory attribute is unknown, or unknown attributes weight more than 50% of the total weights
```

File: docs/dev/explain-personal-search.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+* very_good_match	score >= 75
+* good_match		score >= 50
+* poor_match		score < 50
+* unknown_match		at least one mandatory attribute is unknown, or unknown attributes weights more than 50% of the total weights
+* may_not_match		at least one mandatory attribute score is <= 50 (e.g. may contain traces of an allergen)
+* does_not_match	at least one mandatory attribute score is <= 10 (e.g. contains an allergen, is not vegan)
+
+For the web, this is implemented in `product-search.js`, in function `match_product_to_preferences`
+
+FIXME: why are preferences_factors hardcoded in product-search.js, while we have a preference api
```
Comment (stephanegigandet) [inline docs/dev/explain-personal-search.md:68]: Good catch. I filed https://github.com/openfoodfacts/openfoodfacts-server/issues/10406

**Review by stephanegigandet — APPROVED** [review]
Thanks a lot for adding the documentation

File: docs/api/ref/api.yml (hunk 1 above)
Comment (alexgarel) [reply docs/api/ref/api.yml:477]: I'm fixing the docs to use the short URL instead.

File: html/js/product-search.js (hunk 4 above)
Comment (alexgarel) [reply html/js/product-search.js:20]: changed for a better formulation.

File: docs/dev/explain-personal-search.md (hunk 5 above)
Comment (alexgarel) [reply docs/dev/explain-personal-search.md:68]: Transformed the fixme to a note

**Pull request merged**